From 20f69bc657233ac120a0fd45088930b4f21dc63a Mon Sep 17 00:00:00 2001
From: Tetsutoki Shiozawa <tetsutoki.shiozawa@sony.com>
Date: Wed, 25 Oct 2017 19:17:47 +0900
Subject: [PATCH 035/296] Turning screen on without creating surface

Symptopm:
The screen doesn't wake up when it gets a second in-coming call.

Solution:
If the surface is already visible, TurnOnScreen flag is not
evaluated. We should directly request turning screen on to
the manager. This is the last resort to wakeup screen.
---
 services/core/java/com/android/server/wm/WindowState.java | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index 77a8fec7241..a3c378ddb84 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -2315,8 +2315,12 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
 
         // If we were already visible, skip rest of preparation.
         if (wasVisible) {
-            if (DEBUG_VISIBILITY) Slog.v(TAG,
-                    "Already visible and does not turn on screen, skip preparing: " + this);
+            if (DEBUG_VISIBILITY) Slog.v(TAG, "Already visible, skip preparing: " + this);
+            if (mTurnOnScreen) {
+                // If the surface is already visible, TurnOnScreen flag is not evaluated.
+                // We should directly request turning screen on to the manager.
+                mService.mTurnOnScreen = true;
+            }
             return;
         }
 
-- 
2.11.0

